db/db_impl.h

```diff
@@ -848,14 +851,29 @@ class DBImpl : public DB {
                             bool disable_memtable = false,
                             uint64_t* seq_used = nullptr);
 
-  // batch_cnt is expected to be non-zero in seq_per_batch mode and indicates
-  // the number of sub-patches. A sub-patch is a subset of the write batch that
-  // does not have duplicate keys.
-  Status WriteImplWALOnly(const WriteOptions& options, WriteBatch* updates,
-                          WriteCallback* callback = nullptr,
-                          uint64_t* log_used = nullptr, uint64_t log_ref = 0,
-                          uint64_t* seq_used = nullptr, size_t batch_cnt = 0,
-                          PreReleaseCallback* pre_release_callback = nullptr);
+  // Write only to memtables without joining any write queue
+  Status UnorderedWriteMemtable(const WriteOptions& write_options,
+                                WriteBatch* my_batch, WriteCallback* callback,
+                                uint64_t log_ref, uint64_t seq);
+
+  // Whether the batch requires to be assigned with an order
+  enum AssignOrder { DontAssignOrder, DoAssignOrder };
+  // Whether it requires publishing last sequence or not
+  enum PublishLastSeq { DontPublishLastSeq, DoPublishLastSeq };
```

Comment:
My vote is to use boolean instead of this. Btw, the convention is call kDontPublishLastSeq.

Reply:
Enum is the recommended style: https://google.github.io/styleguide/cppguide.html#Function_Argument_Comments
> Consider changing the function signature to replace a bool argument with an enum argument. This will make the argument values self-describing.

Reply:
If you really want to follow the guide, rather than cherry-pick, you should create a class for arguments:
> If the argument is a literal constant, and the same constant is used in multiple function calls in a way that tacitly assumes they're the same, you should use a named constant to make that constraint explicit, and to guarantee that it holds.
>
> Consider changing the function signature to replace a bool argument with an enum argument. This will make the argument values self-describing.